src/hooks/use-deep-equality.ts

```diff
@@ -0,0 +1,18 @@
+import {
+  useState, useEffect, SetStateAction, Dispatch,
+} from 'react';
+import deepEquals from 'fast-deep-equal';
+
+export default function useDeepEquality<T>(value?: T | undefined): [T| undefined, Dispatch < SetStateAction < T| undefined >>, T| undefined] {
```

Comment:
The spacing seems a little odd in the return type here. Does this work ok instead?
```typescript
export default function useDeepEquality<T>(value?: T | undefined): [T | undefined, Dispatch<SetStateAction<T | undefined>>, T | undefined] {

```